build/yaml/steps/codeanalysis/csharp.yaml

```diff
@@ -9,7 +9,7 @@ steps:
   inputs:
     continueOnError: true
     rulesetName: Custom
-    msBuildArchitecture: x64
-    customRuleset: DevDivRoslynRequired.ruleset
+    msBuildArchitecture: amd64
+    customRuleset: $(Build.SourcesDirectory)\build\CodeAnalysis\Sdl.Recommended.Error.ruleset
```

Comment:
I will revert this back what it was before (even though it is broken) and then fix whatever warnings/errors come out of the RoslynAnalyzers task.

I will leave the changes to use the SDL ruleset during the build phase in order to aid in resolving most of the overlapping rules from the SDL ruleset and the one that this task should be using.

In a later change, I'll figure out how to run the appropriate ruleset with this task.